src/test/java/teammates/logic/core/NotificationsLogicTest.java

```diff
@@ -0,0 +1,107 @@
+package teammates.logic.core;
+
+import java.time.Instant;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.NotificationType;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.storage.api.NotificationsDb;
+
+/**
+ * SUT: {@link NotificationsLogic}.
+ */
+public class NotificationsLogicTest extends BaseLogicTest {
+    private NotificationAttributes n;
+    private final NotificationsDb notifDb = NotificationsDb.inst();
+    private final NotificationsLogic notifLogic = NotificationsLogic.inst();
+
+    @Override
+    protected void prepareTestData() {
+        // test data is refreshed before each test case
+    }
+
+    @BeforeMethod
+    public void refreshTestData() {
+        dataBundle = getTypicalDataBundle();
+        removeAndRestoreTypicalDataBundle();
+    }
+
+    @Test
+    public void testAll() throws Exception {
+        testGetNotification();
+        testCreateNotification();
+    }
```

Comment:
Should add tests for `updateNotification`, `deleteNotification`, `getAllNotifications` and `getActiveNotificationsByTargetUser` as well, you can make use of test data from typical data bundle.

Reply:
I was originally thinking of splitting it into individual PRs for each route so that [PR#11665](https://github.com/TEAMMATES/teammates/pull/1166) can be merged ASAP. Since PR#11665 makes some changes to POST and requires these tests to ensure its correctness.

Reply:
I see, but #11665 seems to be blocked by Notification feature - Database Test #11651 which has pending changes, maybe consider adding `updateNotification` and `deleteNotification` first?

Reply:
Sure thing, does the POST tests look fine as of right now?

Reply:
The rest of this PR looks pretty ok to me, except the `verifyPresentInDatabase` which still kind of bugs me.

Reply:
Agreed, this can be discussed, as the only way I can think of is creating the route for `getNotification(notifId)` purely for testing purposes

Reply:
@zhaojj2209 yes I think it is reasonable too, take note that the current resource URI for the GET endpoint (`ResourceURIs.NOTIFICATION`) will need to be changed to `ResourceURIs.NOTIFICATIONS`. `ResourceURIs.NOTIFICATION` should be for getting single notification.